### PR TITLE
Add model effort writeup to DG

### LINKED ISSUE #259: Add Manual testing instructions for findPool command

### LINKED ISSUE #273: Tweak DG user storeies to reflect final product

### PR DESCRIPTION
**Pull request by ZechariahTan**
* Improve DG user story accuracy to fit current iteration of the product
* Improve Use case formatting
* Add findPool manual testing instructions

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -294,37 +294,43 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (un
...
+|* * *   |HR Executive  |search for specific type of passengers                            |I can see if any passengers fulfil a criteria and view their carpool details                    |
 |* * *   |HR Executive  |list all passengers                                              |I can see all the passengers available                                                          |
-|* * *   |HR Executive  |create a profile                                                 |I can easily manage and track drivers and passengers                                            |
-|* * *   |HR Executive  |delete employee profile                                          |passenger's data will not be stored when they are no longer looking to carpool                  |
-|* *     |HR Executive  |filter passengers' destination and pickup point based on location|drivers are not heavily inconvenienced to pick up passengers                                    |
-|* *     |HR Executive  |match only with female drivers                                   |so that female colleagues looking to carpool only with female drivers can be easily accomodated |
```
Comment (lyueyang) [inline docs/DeveloperGuide.md:374]: how about if we were to say that the passengers are filtered by femaleDrivers only

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -573,12 +589,11 @@ The following section describes the challenges our team faced in this project, t
 
 #### Extension of Model
 
-{Zech to elaborate on the challenges faced when extending the model
-some highlights:
-added new fields
-a great deal of refactoring was done
-model was reworked to contain a passenger and pool list
-the initial person class was refactored to become driver and passenger}
+As our group had initially decided on morphing the existing AB3, changes to the model consisted mainly of refactoring existing classes. Although a rather trivial task, this process was rather time-consuming and tedious. Due to the scale of the existing AB3 code, some difficulty was encountered during this refactoring stage as some making seemingly minute changes in method signatures or names would result in a need to edit other segments of affected code. Although the use of IntelliJ’s toolbox greatly sped up the process, the nature of some of the refactoring changes we intended to make meant that some of said changes were not as easily detected by IntelliJ, resulting in some uncaught bugs following our refactoring stage. The brownfield nature of the project also lead to some difficulty in understanding the model and how the different modules interact with each other.
```
Comment (lyueyang) [inline docs/DeveloperGuide.md]: Maybe 

The brownfield nature of the project also _led_ to some...

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ZechariahTan) [reply docs/DeveloperGuide.md:374]: Yeap ok that would make sense. Ill rephrase the user story.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ZechariahTan) [reply docs/DeveloperGuide.md]: Right my mistake

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -573,12 +589,11 @@ The following section describes the challenges our team faced in this project, t
...
+
+Nearing the end of the first milestone, every team member had a better understanding of the existing code allowing for a better segmentation of work. As this was also the period where new additions were made to the existing AB3 model instead of simple refactoring, there was also less confusion surrounding what components of the model needed to be modified. This was also helped by the fact that in the case of adding features, we had a clearer idea of what needed to be done for the new feature instead of following the trails of an existing one.
```
Comment (chewterence) [inline docs/DeveloperGuide.md]: perhaps "Towards the end of the first milstone"

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (chewterence) [reply docs/DeveloperGuide.md]: i think you could omit the "although a trivial task" beacause the idea is to show that it is not trivial

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -549,15 +556,24 @@ testers are expected to do more *exploratory* testing.
...
+### Finding a pool
 
-### Saving data
+1. Finding a pool while all pools are being shown.
 
-1. Dealing with missing/corrupted data files.
+    1. Prerequisites: Multiple pools in the list. List all pools using the `list` command.
 
-    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+    1. Test case: `findPool n/lenny`.<br>
```
Comment (chewterence) [inline docs/DeveloperGuide.md:638]: is the tester expecting exactly these specific people? `Alan Poh`, `Lenny Hoon`, and `Turner Peck`
What if the tester has other names in the list instead

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -549,15 +556,24 @@ testers are expected to do more *exploratory* testing.
...
+       Expected: Details of only the pool with `Alan Poh`, `Lenny Hoon`, and `Turner Peck` is shown. Status message shows 1 pool listed.
 
-1. _{ more test cases …​ }_
+    1. Test case: `findPool n/michael`.<br>
```
Comment (chewterence) [inline docs/DeveloperGuide.md:641]: what if the tester has a michael in the list?

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ZechariahTan) [reply docs/DeveloperGuide.md:638]: These tests are provided on the assumption that the user is testing with the default addressbook.json that is created by the GME system. This also addresses your other comment regarding the test case. I will fix this by adding this assumption in the prerequisites section of the testcases.

**Pull request merged**